cmd/kaspactl/main.go

```diff
@@ -23,28 +31,56 @@ func main() {
 	}
 	defer client.Disconnect()
 
-	var responseString string
-	done := make(chan struct{})
+	responseChan := make(chan string)
 
-	go func() {
-		requestString := cfg.RequestJSON
-		var err error
-		responseString, err = client.PostJSON(requestString)
-		if err != nil {
-			printErrorAndExit(fmt.Sprintf("error posting the request to the RPC server: %s", err))
-		}
-		done <- struct{}{}
-	}()
+	if cfg.RequestJSON != "" {
+		go postJSON(cfg, client, responseChan)
+	} else {
+		go postCommand(cfg, client, responseChan)
+	}
 
-	timeout := time.Duration(cfg.Timeout) * time.Second
+	timeout := time.Duration(cfg.Timeout) * time.Second * 10000 // TODO: remove the * 10000
 	select {
-	case <-done:
+	case responseString := <-responseChan:
 		fmt.Println(responseString)
 	case <-time.After(timeout):
 		printErrorAndExit(fmt.Sprintf("timeout of %s has been exceeded", timeout))
 	}
 }
 
+func printAllCommands() {
+	requestDescs := commandDescriptions()
+	for _, requestDesc := range requestDescs {
+		fmt.Printf("\t%s\n", requestDesc.help())
+	}
+}
+
+func postCommand(cfg *configFlags, client *grpcclient.GRPCClient, responseChan chan string) {
+	message, err := parseCommand(cfg.CommandAndParameters, commandDescriptions())
+	if err != nil {
+		printErrorAndExit(fmt.Sprintf("error parsing command: %+v", err))
```

Comment:
I don't think you want to print the stack trace here.

Reply:
I'm not sure about that... I printed the stack trace precisely because I needed it.

I think a real solution is something like RuleError that is returned when the parameters are wrong, and you print without stack trace, and any other error that you print wholesale.

But for now removed the %+v to be similar to the other errors in this area.